flow/instrumentation/setup.go

```diff
@@ -0,0 +1,40 @@
+package instrumentation
+
+import (
+	"context"
+	"errors"
+
+	"github.com/PeerDB-io/peer-flow/instrumentation/tracing"
+)
+
+type Config struct {
+	EnableTracing bool
+}
+
+func SetupInstrumentation(ctx context.Context, serviceName string, config Config) (func(ctx context.Context) error, error) {
+	var shutdownFuncs []func(context.Context) error
+	shutdown := func(ctx context.Context) error {
+		var err error
+		for _, fn := range shutdownFuncs {
+			err = errors.Join(err, fn(ctx))
+		}
+		shutdownFuncs = nil
+		return err
+	}
+
+	var err error
+	handleErr := func(inErr error) {
```

Comment:
shouldn't be a function

Reply:
added as we'll setup more things in this